x/stakeibc/keeper/transfer.go

```diff
@@ -0,0 +1,102 @@
+	ibctypes "github.com/cosmos/ibc-go/v7/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+
+	"github.com/Stride-Labs/stride/v14/x/stakeibc/types"
+)
+
+// Transfers tokens from the community pool deposit ICA account to the host zone holding address for that pool
+func (k Keeper) IBCTransferCommunityPoolTokens(ctx sdk.Context, token sdk.Coin, communityPoolHostZone types.HostZone, shouldLiquidStake bool) error {
+
+	// The memo may contain autopilot commands to atomically liquid stake tokens when transfer succeeds
+	//  both transfer+liquid stake will succeed and stTokens will end in the stride side holding address, 
+	//  or neither will and the original base tokens will return to the foreign deposit ICA address
+	memoCommands := ""
+	autopilotCmd := "{ \"autopilot\": { \"receiver\": \"%s\",  \"stakeibc\": { \"action\": \"LiquidStake\" } } }"
+	if shouldLiquidStake {
+		memoCommands = fmt.Sprintf(autopilotCmd, communityPoolHostZone.CommunityPoolHoldingAddress)
+		stakedDenom := types.StAssetDenomFromHostZoneDenom(token.Denom)
+		k.Logger(ctx).Info(fmt.Sprintf("[IBCTransferCommunityPoolTokens] Transferring %v %s and then liquid staking to %s", token.Amount, token.Denom, stakedDenom))
+	} else {
+		k.Logger(ctx).Info(fmt.Sprintf("[IBCTransferCommunityPoolTokens] Transferring %v %s", token.Amount, token.Denom))
+	}
+	
+	// NOTE: this assumes no clock drift between chains, which tendermint guarantees
+	ibcTransferTimeoutNanos := k.GetParam(ctx, types.KeyIBCTransferTimeoutNanos)
+	timeoutTimestamp := uint64(ctx.BlockTime().UnixNano()) + ibcTransferTimeoutNanos
+
+	msg := ibctypes.NewMsgTransfer(
+		ibctypes.PortID,
+		communityPoolHostZone.TransferChannelId,
+		token,
+		communityPoolHostZone.CommunityPoolDepositIcaAddress, // ICA controlled address on foreign hub
+		communityPoolHostZone.CommunityPoolHoldingAddress, // Stride address, unique to each community pool / hostzone
+		clienttypes.Height{},
+		timeoutTimestamp,
+		memoCommands,
+	)
+
+	// Submit IBC transfer msg
+	msgTransferResponse, err := k.RecordsKeeper.TransferKeeper.Transfer(sdk.WrapSDKContext(ctx), msg)
+	if err != nil {
+		return err
+	}
+
+	k.Logger(ctx).Info(fmt.Sprintf("[IBCTransferCommunityPoolTokens] Successfully submitted ibc transfer message %+v", msgTransferResponse))
+
+	return nil
+}
+
+
+// Transfers all tokens in the Stride-side holding address over to the communityPoolReturnAddress ICA
+func (k Keeper) IBCReturnAllCommunityPoolTokens(ctx sdk.Context, communityPoolHostZone types.HostZone) error {
+	// Use bankKeeper to see all coin types and amounts currently in the stride-side holding address
+	req := banktypes.NewQueryAllBalancesRequest(sdk.AccAddress(communityPoolHostZone.CommunityPoolHoldingAddress), nil)
+	resp, err := k.bankKeeper.AllBalances(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	memo := ""
+	errors := make([]error, 0)
+
+	for _, foundCoin := range resp.Balances {
```

Comment:
For this particular transfer which is from stride "holding" module account back to the return ICA, we don't need any token type check because all tokens which are found in the holding account should go back to the return ICA, and eventually get fund-community-pool called on them regardless of their denom.

Tokens end up in the holding address after a transfer and possible autopilot action (stake, redeem, or no action).  So there could be the native denom, the stNativeDenom, or other weird 3rd party denoms (or all of these possibilities at once) in the holding address and it doesn't matter because they all are going back to the return ICA and then the pool each stride epoch.

Reply:
makes sense, thanks for the explanation!